### ISSUE TITLE
"align" rule "parameters" shows unexpected error, when parameter is aligned using tabs (not spaces)

### ISSUE THREAD

**Issue by atsu85** [palantir/tslint#3636]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.2
- __Running TSLint via__: CLI

#### TypeScript code being linted
following code is based on test [invalidParametersAlignment3](https://github.com/palantir/tslint/blob/master/test/rules/align/parameters/test.ts.fix), but parameters `shouldBeOkWhenTabWidthIs4`, `shouldBeOkWhenTabWidthIs4` and `isOkWhenTabWidthIs1` use tabs instead of spaces for alignment.
```ts
// uses as many tabs as needed to align with first parameter
// (no extra spaces needed in this case, as in this case alignToColumn==36 and if tab width is 4, then `alignToColumn % tabWidth === 0`)
function invalidParametersAlignment(ok1: number,
                                    ok2: number, // OK: aligned with spaces
									shouldBeOkWhenTabWidthIs4: number, // NOT OK: VISUALLY aligned when IDE (unless tab is configured to take same with as single space)
									// ~~~~~~~~~  [parameters are not aligned]
																		shouldBeOkWhenTabWidthIs2: number, // NOT OK: VISUALLY aligned when IDE (unless tab is configured to take same width as single space)
																		// ~~~~~~~~~  [parameters are not aligned]
																																				isOkWhenTabWidthIs1: number // OK: NOT VISUALLY aligned (unless IDE is configured to use same with for tab as for single space)
) {
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "align": [true, "parameters"]
  }
}
```

#### Actual behavior
Currently the failure is added [based on parameter character index in the source file line](https://github.com/palantir/tslint/blob/b5ebb9fc3d8cf6fa2f1ec02beb233d02a60ffbf6/src/rules/alignRule.ts#L168-L169), so when using tabs instead of spaces for formatting the code, You'd need to add as many tabs as there would need to be spaces - as a result to obey the align (and indent) rule, the code isn't visually aligned (unless the tab size is configured in editor to be displayed as narrow as single character, instead of smth more common 2 or 4 or 8 characters)

#### Expected behavior
It would be super-cool if align rule could access and use configuration of indent rule to be able to calculate visual width for alignment. I suspect it isn't possible right now, is it?
As a easy to implement alternative: allow specifying tab visual width with align rule argument (probably should default to 1, to avoid breaking changes).

**Comment by atsu85** [comment]
I'd be happy to implement it, but 
1) it would be nice if someone (perhaps @ajafff, @adidahiya) who isn't total stranger to this project, could confirm that it isn't possible to read configuration parameters of another rule.

2) When implementing with alternative approach:
> specifying tab visual width with align rule argument 

what would You suggest as a argument name? Perhaps `tabWidth` or `tabVisualWidth`?

**Comment by JoshuaKGoldberg** [comment]
> confirm that it isn't possible to read configuration parameters of another rule

Correct, this is an intentional design choice to keep rules simpler. If rule A creates a suggested fix that results in code violating rule B, applying later suggested fixes from rule B should result in code without complaints.

The root problem here is that you're mixing tab indentation with character indentation. Your tabs declare how much lines start indenting, then code characters within the lines increase that indent. 

The only way the `align` rule can sanely suggest fixes here without improperly relying on mixing space types would be to push the first parameter onto the next line:

```typescript
// uses as many tabs as needed to align with first parameter
// (no extra spaces needed in this case, as in this case alignToColumn==36 and if tab width is 4, then `alignToColumn % tabWidth === 0`)
function invalidParametersAlignment(
    ok1: number,
    ok2: number,
    shouldBeOkWhenTabWidthIs4: number,
    shouldBeOkWhenTabWidthIs2: number,
) {
}
```

Anecdotally, I find the current behavior of pushing everything forward very annoying when turning the rule on and auto-fixing across existing codebases, so +1 to this fix. 😉

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
